lib/segment/src/data_types/primitive.rs

```diff
@@ -67,6 +68,61 @@ impl PrimitiveVectorElement for VectorElementType {
     }
 }
 
+impl PrimitiveVectorElement for VectorElementTypeHalf {
+    fn slice_from_float_cow(vector: Cow<[VectorElementType]>) -> Cow<[Self]> {
+        Cow::Owned(vector.iter().map(|&x| f16::from_f32(x)).collect())
```

Comment:
you convert `f16-vectors` several times. I propose to define a separate function for `f16`-vector conversions